lightning-persister/src/manager.rs

```diff
@@ -0,0 +1,219 @@
+use lightning::chain;
+use lightning::chain::chaininterface::{BroadcasterInterface, FeeEstimator};
+use lightning::chain::keysinterface::InMemoryChannelKeys;
+use lightning::ln::channelmanager::{SimpleArcChannelManager};
+use lightning::util::logger::Logger;
+use lightning::util::ser::Writeable;
+use super::DiskWriteable;
+use std::fs;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use std::thread;
+
+#[cfg(not(target_os = "windows"))]
+use std::os::unix::io::AsRawFd;
+
+impl<M: chain::Watch<Keys=InMemoryChannelKeys>, T: BroadcasterInterface, F: FeeEstimator, L: Logger> DiskWriteable
+	for SimpleArcChannelManager<M, T, F, L> {
+	fn write_to_file(&self, writer: &mut fs::File) -> Result<(), std::io::Error> {
+		self.write(writer)
+	}
+}
+
+pub struct ManagerPersister {
+	#[cfg(test)]
+	path_to_manager: String,
+}
+
+impl ManagerPersister {
+	pub fn start<M: 'static + chain::Watch<Keys=InMemoryChannelKeys>, T: 'static + BroadcasterInterface, F: 'static + FeeEstimator, L: 'static + Logger>(filepath: String, filename: String, manager: SimpleArcChannelManager<M, T, F, L>) -> Self {
+		#[cfg(test)]
+		let saved_filepath = filepath.clone();
+
+		let mutcond = Arc::clone(&manager.persistence_lock);
+		thread::spawn(move || {
```

Comment:
Since we're gonna eat the cost and just run a full thread here lets also call `ChannelManager::timer_chan_freshness_every_min()` once a minute.